src/arch/arm/kernel/boot.c

```diff
@@ -274,6 +277,11 @@ BOOT_CODE static bool_t try_init_kernel_secondary_core(void)
 #endif /* CONFIG_ARM_HYPERVISOR_SUPPORT */
     NODE_LOCK_SYS;
 
+#if defined(CONFIG_KERNEL_MCS) && defined(CONFIG_DEBUG_BUILD)
+    clock_sync_test[getCurrentCPUIndex()] = getCurrentTime();
+    /* Make sure clock_sync_test is set before ksNumCPUs */
+    __atomic_thread_fence(__ATOMIC_ACQ_REL);
```

Comment:
A barrier that ensures writes must finish should sufficient here,
```suggestion
        /* Ensure the update of clock_sync_test is visible before the update of ksNumCPUs */
        __atomic_thread_fence(__ATOMIC_RELEASE);
```